File: SpringBoot/src/main/java/com/hospital/qlbv/entity/Appointment.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "appointments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer appointmentId;
    
    @Column(name = "patient_id")
    private Integer patientId;
    
    @Column(name = "doctor_id")
    private Integer doctorId;
    
    @Column(name = "department_id")
    private Integer departmentId;
    
    @Column(name = "appointment_date")
    private String appointmentDate;
    
    @Column(name = "time_slot")
    private String timeSlot;
    
    private String status;
    private String notes;
    private String reason;
    
    @Column(name = "appointment_type")
    private String appointmentType;
    
    @Column(name = "is_emergency")
    private Boolean isEmergency;
    
    private Integer priority;
    
    @Column(name = "created_at")
    private String createdAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"user", "insurance", "appointments"})
    private Patient patient;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"user", "department", "room", "appointments"})
    private Doctor doctor;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"doctors", "rooms"})
    private Department department;
}

File: Backend/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class Appointment
{
    [Key]
    public int AppointmentID { get; set; }
    public int? PatientID { get; set; }
    public int? DoctorID { get; set; }
    public int? DepartmentID { get; set; }
    public DateTime? AppointmentDate { get; set; }
    public string? TimeSlot { get; set; }
    public string? Status { get; set; }
    public string? Notes { get; set; }
    public string? Reason { get; set; }
    public string? AppointmentType { get; set; }
    public bool? IsEmergency { get; set; }
    public int? Priority { get; set; }
    public DateTime? CreatedAt { get; set; }

    // Navigation properties
    public virtual Patient? Patient { get; set; }
    public virtual Doctor? Doctor { get; set; }
    public virtual Department? Department { get; set; }
}
